lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -1319,26 +1319,35 @@ sub lookupFontinfo {
   #  return LookupValue(($defn ? ToString($defn) : ToString($token)) . '_fontinfo'); }
   return LookupValue('fontinfo_' . ($defn ? $defn->getCSName : ToString($token))); }
 
-# Need to handle "at" too!!!
+# This should eventually actually load the font metrics,
+# and tie-in to the FontMetrics data used by Font.
 DefPrimitive('\font Token SkipMatch:= SkipSpaces TeXFileName', sub {
     my ($stomach, $cs, $name) = @_;
     my $gullet = $stomach->getGullet;
     $name = ToString($name);
+    my ($at, $scaled);
+    if ($gullet->readKeyword('at'))     { $at     = $gullet->readDimension; }
+    if ($gullet->readKeyword('scaled')) { $scaled = $gullet->readNumber; }
     my %props = LaTeXML::Common::Font::decodeFontname($name,
-      ($gullet->readKeyword('at')
-        ? $gullet->readDimension->ptValue : undef),
-      ($gullet->readKeyword('scaled')
-        ? $gullet->readNumber->valueOf / 1000 : undef));
+      $at && $at->ptValue, $scaled && $scaled->valueOf / 1000);
     if (!keys %props) {    # Failed?
       Info('unexpected', $name, $stomach, "Unrecognized font name '$name'",
         "Font switch macro " . ToString($cs) . " will have no effect"); }
     else {
       $props{fontname} = $name; }
+    my $f = ($at ? $at->divide(Dimension('1em'))->valueOf
+      : ($scaled ? $scaled->valueOf / 1000
+        : 1));
+    my $fontinfo = {%props};
```

Comment:
Out of curiosity, why not `my $fontinfo = \%props;` ?
I think that avoids cloning the hash, and if I'm reading the code right `%props` has no further uses that may cause data race conflicts.